From 1e48585a6faab0ab24c89b84ae79fd741510a2fb Mon Sep 17 00:00:00 2001
From: Auke Kok <auke-jan.h.kok@intel.com>
Date: Thu, 12 Jul 2012 21:50:07 -0700
Subject: [PATCH] bugfix: incorrect sizeof() usage for memset.

This ends up memsetting the size of a pointer instead of the
entire struct.

Signed-off-by: Auke Kok <auke-jan.h.kok@intel.com>
---
 libnetlink.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libnetlink.c b/libnetlink.c
index cfc5f15..e61d417 100644
--- a/libnetlink.c
+++ b/libnetlink.c
@@ -41,7 +41,7 @@ int rtnl_open_byproto(struct rtnl_handle *rth, unsigned subscriptions,
 	int sndbuf = 32768;
 	int rcvbuf = 32768;
 
-	memset(rth, 0, sizeof(rth));
+	memset(rth, 0, sizeof(struct rtnl_handle));
 
 	rth->fd = socket(AF_NETLINK, SOCK_RAW | SOCK_CLOEXEC, protocol);
 	if (rth->fd < 0) {
-- 
2.15.0

